### PR TITLE
Bump testcontainers-java-tarantool to 1.2.0

### LINKED ISSUE #442: Bump testcontainers-java to 1.2.0

### PR DESCRIPTION
**Pull request by nickkkccc**
***Note the peculiarity*** of instance restarts in a cluster [#106](https://github.com/tarantool/testcontainers-java-tarantool/issues/106)

I haven't forgotten about:
- [x] Tests
- [x] Changelog
- [ ] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:
Closes #442

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -264,7 +264,7 @@
         <dependency>
             <groupId>io.tarantool</groupId>
             <artifactId>testcontainers-java-tarantool</artifactId>
-            <version>1.0.1</version>
+            <version>1.0.2</version>
```
Comment (dkasimovskiy) [inline pom.xml:267]:
```suggestion
            <version>1.1.0</version>
```

**Review by dkasimovskiy — CHANGES_REQUESTED** [review]

**Review by dkasimovskiy — APPROVED** [review]

File: src/test/resources/cartridge/init.lua [hunk 2]
```diff
@@ -16,7 +16,7 @@ local ok, err = cartridge.cfg({
         'app.roles.api_storage',
         'app.roles.custom',
     },
-    cluster_cookie = 'testapp-cluster-cookie',
+    cluster_cookie = 'secret-cluster-cookie',
```
Comment (ArtDu) [inline src/test/resources/cartridge/init.lua:19]: it is default value and you can just remove this whole line

File: src/test/java/io/tarantool/driver/integration/ssl/SslClientITEnterprise.java [hunk 3]
```diff
@@ -56,7 +56,6 @@ public static void setUp() throws URISyntaxException, SSLException {
             .withUsername("test_user")
             .withPassword("test_password")
             .withMemtxMemory(256 * 1024 * 1024)
-            .withDirectoryBinding(RESOURCE_PATH + "ssl")
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/ssl/SslClientITEnterprise.java:59]: you won't get files that are placed in `org/testcontainers/containers/enterprise/ssl` that needed for testing

**Review by ArtDu — CHANGES_REQUESTED** [review]

File: src/test/java/io/tarantool/driver/integration/ssl/SslClientITEnterprise.java (hunk 3 above)
Comment (nickkkccc) [reply src/test/java/io/tarantool/driver/integration/ssl/SslClientITEnterprise.java:59]: Fixed it for enterprise tests

File: src/test/resources/cartridge/init.lua (hunk 2 above)
Comment (nickkkccc) [reply src/test/resources/cartridge/init.lua:19]: Removed

**Review by bitgorbovsky — APPROVED** [review]

File: src/test/java/io/tarantool/driver/integration/CartridgeMixedInstancesContainer.java [hunk 4]
```diff
@@ -23,7 +23,6 @@ abstract class CartridgeMixedInstancesContainer {
                         "cartridge-java-test-mixed",
                         "cartridge/instances_mixed.yml",
                         "cartridge/topology_mixed.lua")
-                        .withDirectoryBinding("cartridge")
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/CartridgeMixedInstancesContainer.java:26]: I don't get how this works. Please explain how does a container get needed files?

**Review by ArtDu — COMMENTED** [review]
LGTM, but I have a question about container behavior

File: src/test/java/io/tarantool/driver/integration/CartridgeMixedInstancesContainer.java (hunk 4 above)
Comment (nickkkccc) [reply src/test/java/io/tarantool/driver/integration/CartridgeMixedInstancesContainer.java:26]: ***Previous version of the test containers (e.g., 1.0.1):***
1. Dependencies are downloaded on the system inside the container (gcc and others).
2. The cartridge folder is a mapping of the `/app` directory.
3. The cartridge folder is copied into the image as the `/app` working directory. This step is where the image is built.
4. Start the container and run the `cartridge build` command (dependency rocks are downloaded into the working directory at this point).
5. `Key point:` the dependency rocks are downloaded into the `/app` directory related to the container. But docker looks at the cartridge folder that was mapped earlier in the build phase.

***New version of testcontainers:***

Copying the `/cartridge` directory and executing cartridge build happens during the image build phase. `Key point:` when using the `.withDirectoryBinding` method at the container startup stage, docker will look at the `/cartridge` directory that doesn't have the built rocks dependencies in it

**Review by ArtDu — APPROVED** [review]

**Pull request merged**